File: app/src/main/java/com/example/lenovo/reader/fragments/dashboard/DashboardModule.java
package com.example.lenovo.reader.fragments.dashboard;

import com.example.lenovo.reader.annotations.PerFragment;
import dagger.Module;
import dagger.Provides;

@Module()
public class DashboardModule {

    private DashboardFragment dashboardFragment;
    private DashboardContract.DashboardPresenter dashboardPresenterImpl;

    public DashboardModule(DashboardFragment dashboardFragment, DashboardPresenterImpl dashboardPresenterImpl) {
        this.dashboardFragment = dashboardFragment;
        this.dashboardPresenterImpl = dashboardPresenterImpl;
    }

    @Provides @PerFragment
    DashboardContract.DashboardView provideDashboardView() {
        return dashboardFragment;
    }

    @Provides @PerFragment
    DashboardContract.DashboardPresenter provideDashboardPresenter() {
        return dashboardPresenterImpl;
    }
}

File: app/src/main/java/com/example/lenovo/reader/fragments/dashboard/DashboardModule.kt
package com.example.lenovo.reader.fragments.dashboard

import com.example.lenovo.reader.annotations.FragmentScope
import com.example.lenovo.reader.fragments.dashboard.interactors.GetLastAddedArticlesInteractor
import com.example.lenovo.reader.fragments.dashboard.interactors.GetLastAddedArticlesInteractorImpl
import dagger.Binds
import dagger.Module

@Module
abstract class DashboardModule {

  @Binds
  @FragmentScope
  abstract fun provideDashboardView(dashboardFragment: DashboardFragment): DashboardView

  @Binds
  @FragmentScope
  abstract fun provideDashboardPresenter(dashboardPresenterImpl: DashboardPresenterImpl): DashboardPresenter

  @Binds
  @FragmentScope
  abstract fun provideGetLastAddedArticlesInteractor(getLastAddedArticlesInteractorImpl: GetLastAddedArticlesInteractorImpl): GetLastAddedArticlesInteractor


}
